cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -567,19 +598,61 @@ open class Node @Inject constructor(private val project: Project) {
         if (!config.hasPath("sshd.port")) {
             sshdPort(defaultSsh)
         }
-        val configDefaults = ConfigFactory.empty()
-            .withValue("dataSourceProperties.dataSource.url", ConfigValueFactory.fromAnyRef("jdbc:h2:file:./persistence/persistence;DB_CLOSE_ON_EXIT=FALSE;WRITE_DELAY=0;LOCK_TIMEOUT=10000"))
+
+        // TODO Refactor this code to extract the information from the config file
+        val p2pHost = config.getString("p2pAddress")
```

Comment:
I'm a bit confused with what's going on in this line and below to line 608. I think this should just work the way it did before, so why do you need to do this extra address parsing? If this is required, it should probably be factored out as you're effectively doing the same thing three times (finding out what the port number is from the address string).

Reply:
This code was added to read the p2p/rpc/adminrpc port numbers from the build.gradle. Before it was working fine as long as there was only one Corda node container in the network. This was because the rpc and p2p port numbers from the build.gradle file were ignored, and hardcoded values were used instead. Using the same port number by more than one service in the same network was triggering a port already allocated error, and the Corda node container would shut down.